dotCMS/src/main/java/com/dotmarketing/portlets/structure/action/EditFieldAction.java

```diff
@@ -318,6 +319,43 @@ private boolean _saveField(ActionForm form, ActionRequest req, ActionResponse re
                  }
             }
 
+            // Validate values entered for decimal/number type check box field
+            if (field.getFieldType().equals(FieldType.RADIO.toString())){
+                String values = fieldForm.getValues();
+                String temp = values.replaceAll("\r\n","|");
+                String[] tempVals = StringUtil.split(temp.trim(), "|");
+
+                if (tempVals.length % 2 == 1){
+                    SessionMessages.add(req, "error", "message.structure.missingdatavalue");
+                    return false;
+                }
+
+                try {
+                    if(dataType.equals(Field.DataType.FLOAT.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            Float.parseFloat(tempVals[i]);
+                        }
+                    }else if(dataType.equals(Field.DataType.INTEGER.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            Integer.parseInt(tempVals[i]);
+                        }
+                    }else if(dataType.equals(Field.DataType.BOOL.toString())){
+                        for(int i=1;i<tempVals.length;i+= 2){
+                            if(!isDBBoolean(tempVals[i])){
+                                String message = "message.structure.invaliddataboolean";
+                                SessionMessages.add(req, "error", message);
+                                return false;
+                            }
+                        }
+                    }
+
+                }catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)